.github/workflows/run-tests.yaml

```diff
@@ -59,5 +63,13 @@ jobs:
       uses: codecov/codecov-action@v3
       with:
         token: ${{ secrets.CODECOV_TOKEN }}
+        flags: ${{ matrix.os }},${{ matrix.python-version }}
         fail_ci_if_error: true
         files: test_artifacts/cobertura.xml
+
+    - name: Create artifacts
+      uses: actions/upload-artifact@v3
+      if: ${{ !cancelled() }}
+      with:
+        name: test-artifacts_${{ matrix.os }}_${{ matrix.python-version }}
+        path: test_artifacts/
```

Comment:
Probably need to specify for how long it's kept

Reply:
[Default: 90 days](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#:~:text=By%20default%2C%20GitHub%20stores%20build%20logs%20and%20artifacts%20for%2090%20days%2C%20and%20this%20retention%20period%20can%20be%20customized).

Unless that incurs costs for you, "don't care" 😅